posts/2021/09/hypothesis-array-api.ipynb

```diff
@@ -0,0 +1,940 @@
+    "catching bugs before your users do."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "654396e0",
+   "metadata": {},
+   "source": [
+    "## Before we begin\n",
+    "\n",
+    "Hypothesis shipped with its Array API strategies in [version 6.21](https://hypothesis.readthedocs.io/en/latest/changes.html#v6-21-0).\n",
+    "We also need to use NumPy >= 1.22\n",
+    "so that we can test with its\n",
+    "recently merged\n",
+    "Array API implementation—this hasn't been released just yet,\n",
+    "so I would recommend installing a [nightly build](https://anaconda.org/scipy-wheels-nightly/numpy).\n",
+    "\n",
+    "I will be using\n",
+    "the excellent [ipytest](https://github.com/chmp/ipytest/) extension\n",
+    "to nicely run tests in Jupyter\n",
+    "as if we were using [pytest](https://github.com/pytest-dev/pytest/) proper.\n",
+    "For pretty printing I use the superb [Rich](https://github.com/willmcgugan/rich) library,\n",
+    "where I simply override Python's builtin `print`\n",
+    "with [`rich.print`](https://rich.readthedocs.io/en/stable/reference/init.html#rich.print).\n",
+    "I also suppress all warnings\n",
+    "for convenience's sake."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "c4a3cb4a-c6db-442b-9507-bedd03aabbe0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%capture\n",
+    "!pip install hypothesis>=6.21\n",
+    "!pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "43dd3731-a8d2-47b3-b88e-10eddc645f34",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%capture\n",
+    "!pip install ipytest\n",
+    "import ipytest; ipytest.autoconfig(display_columns=80)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "4c7c2561-0671-4763-b5e0-935a1957b68b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%capture\n",
+    "!pip install rich\n",
+    "from rich import print"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "2f5d822b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import warnings; warnings.filterwarnings(\"ignore\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "06545d86-904a-44ca-83c5-d5304a251d55",
+   "metadata": {},
+   "source": [
+    "## What the Array API enables\n",
+    "\n",
+    "The [API](https://data-apis.org/array-api/latest/) standardises functionality of array libraries, which has [numerous benefits](https://data-apis.org/array-api/latest/use_cases.html) for both developers and users.\n",
+    "If you've used NumPy before you'll feel right at home!\n",
+    "\n",
+    "The most exciting prospect for me\n",
+    "is being able to easily write an array-consuming method\n",
+    "that works all the adopting libraries.\n",
+    "Let's try writing this method\n",
+    "to calculate the cumulative sums of an array—"
```

Comment:
I think the dash at the end might be confusing, I'd either replace it with a ":" or a full stop.